### PR TITLE
Download api

### PR DESCRIPTION
**Pull request by KerstenBreuer**
First skeleton for file download.

### CODE REVIEW

**Comment by KerstenBreuer** [comment]
This PR contains:
- addition of download API endpoints
- addition of default metadatasets, files, and submissions to the test db

**Review by MoritzHahn1337 — APPROVED** [review]

File: datameta/api/download.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    Usage: /download/{download_token}
+    """
+    token = request.matchdict['token']
+    hashed_token = security.hash_token(token)
+
+    # get download token from db
+    db = request.dbsession
+    db_token = db.query(models.DownloadToken).filter(
+        models.DownloadToken.value==hashed_token
```
Comment (lkuchenb) [inline datameta/api/download.py:73]: Missing check for expiration, suggestion requires importing `sqlalchemy.and_`
```suggestion
        and_(
            models.DownloadToken.value==hashed_token,
            models.DownloadToken.expires>datetime.now()
        )
```

File: tests/integration/test_file_download.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            assert "Location" in header_dict
+            redirect_url = header_dict["Location"]
+
+            # retrieve file:
+            file_response = self.testapp.get(redirect_url, status=200)
+            
+            # check if file response matched expectations:
+            file_content = file_response.body
+            file_checksum = hashlib.md5(file_content).hexdigest()
+            assert file_checksum==file.checksum
```
Comment (lkuchenb) [inline tests/integration/test_file_download.py:62]: Add check for expired URLs to yield 404
```suggestion
    @parameterized.expand(
        [
            #("name of default file", "name of default user", expires, status_get_url, status_get_file)
            # file-owning user requests submitted file:
            ("default_file_1", "user_a", 1, 307, 200),
            # file-owning user requests submitted file and downloads with expired URL:
            ("default_file_1", "user_a", -1, 307, 404),
            # user of file-owning group requests submitted file:
            ("default_file_1", "user_b", 1, 307, 200),
            # file-owning user requests non-submitted file:
            ("default_file_5", "user_a", 1, 307, 200),
            # user of file-owning group requests non-submitted file:
            ("default_file_5", "user_b", 1, 403, None),
            # user not in file-owning group requests submitted and not submitted files:
            ("default_file_1", "user_c", 1, 403, None),
            ("default_file_5", "user_c", 1, 403, None),
            # side admin not in file-owning group requests submitted and not submitted files:
            ("default_file_1", "admin", 1, 307, 200),
            ("default_file_5", "admin", 1, 307, 200),
            # group admin not in file-owning group requests submitted and not submitted files:
            ("default_file_1", "group_y_admin", 1, 403, None),
            ("default_file_5", "group_y_admin", 1, 403, None),
        ]
    )
    def test_rpc_get_file_url(
        self,
        file_site_id:str,
        user_name:str,
        expires:int,
        status_get_url:int=307,
        status_get_file:int=200
    ):
        user = self.default_users[user_name]
        file = self.default_files[file_site_id]

        response = self.testapp.get(
            base_url + f"/rpc/get-file-url/{file.site_id}?expires={expires}",
            headers=user.auth.header,
            status=status_get_url
        )
       
        if status_get_url==307:
            # follow redirect:
            header_dict = {key: value for key, value in response.headerlist}
            assert "Location" in header_dict
            redirect_url = header_dict["Location"]

            # retrieve file:
            file_response = self.testapp.get(redirect_url, status=status_get_file)
           
            if status_get_file==200:
                # check if file response matched expectations:
                file_content = file_response.body
                file_checksum = hashlib.md5(file_content).hexdigest()
                assert file_checksum==file.checksum
```

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thanks a lot, looks good and works great with the View UI, I'll open an extra PR for that!

Minor change requests below

**Review by lkuchenb — APPROVED** [review]
👍

**Comment by lkuchenb** [comment]
Thanks, feel free to merge once CI is done

**Pull request merged**